### PR TITLE
Add e2e tests.

### PR DESCRIPTION
**Pull request by PhilippSiegmantel**
These tests are gonna be the foundation of a complete testing strategy.
We started with e2e tests, since they give you the most coverage for the
least amount of effort.

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "//": "This file will be recreated with sample data everytime you open cypress, so we leave it 'empty'."
```
Comment (BenediktSeidl) [inline cypress/fixtures/example.json:2]: do we need this file?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      GF_SECURITY_ADMIN_PASSWORD: password
+      GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS: tribe-29-checkmk-datasource
+    ports:
+      - '3000:3000'
+    depends_on:
+      - checkmk
+  checkmk:
+    image: checkmk/check-mk-free:2.1.0p9
+    ports:
+      - '12345:5000'
```
Comment (BenediktSeidl) [inline docker-compose.yml:21]: this is very subjective, but i would prefer `5001` as an external port. if you like `12345` i'm also ok with that.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      GF_ANALYTICS_CHECK_FOR_UPDATES: 'false'
+      GF_SECURITY_ADMIN_USER: admin
+      GF_SECURITY_ADMIN_PASSWORD: password
+      GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS: tribe-29-checkmk-datasource
+    ports:
+      - '3000:3000'
+    depends_on:
+      - checkmk
+  checkmk:
+    image: checkmk/check-mk-free:2.1.0p9
```
Comment (BenediktSeidl) [inline docker-compose.yml:19]: should we pin the versions, or always use the latest version in order to run into problems?
if we use `:latest` for the checkmk tag, perhaps we should also create a weekday job to run the e2e action in order to early detect problems.

**Review by BenediktSeidl — COMMENTED** [review]
looks good to me. some comments from my side, some are very subjective, so feel free to discuss instead of changing it :-)

File: cypress/support/e2e.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands';
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
[the remark this answers is not in the document]
Comment (PhilippSiegmantel) [reply cypress/support/e2e.ts:20]: We will need this soon.

File: cypress.env.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "grafanaUsername": "admin",
+  "grafanaPassword": "password",
+  "cmkUrl": "http://checkmk:5000/cmk",
```
[the remark this answers is not in the document]
Comment (PhilippSiegmantel) [reply cypress.env.json:4]: Rename to `cypressToCmkUrl` and `grafanaToCmkUrl`

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+version: '3'
+
+services:
+  grafana:
+    image: grafana/grafana-oss:9.1.0-ubuntu
+    volumes:
+      - .:/var/lib/grafana/plugins/grafana-checkmk-datasource
+    environment:
+      GF_ANALYTICS_REPORTING_ENABLED: 'false'
+      GF_ANALYTICS_CHECK_FOR_UPDATES: 'false'
```
[the remark this answers is not in the document]
Comment (PhilippSiegmantel) [reply docker-compose.yml:10]: Meaning gf_analytics_check_for_plugin_updates

File: docker-compose.yml (hunk 3 above)
Comment (PhilippSiegmantel) [reply docker-compose.yml:19]: We should use `2.1-latest` and grafana `9-latest` or what ever there is.

File: cypress/fixtures/example.json (hunk 1 above)
Comment (PhilippSiegmantel) [reply cypress/fixtures/example.json:2]: It isn't added anymore every time you open cypress, so it's not necessary anymore.

File: docker-compose.yml (hunk 3 above)
Comment (PhilippSiegmantel) [reply docker-compose.yml:19]: Sadly there is only 'main' and 'latest' for Grafana besides specific versions.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -22,3 +22,9 @@ jobs:
 
       - name: Build and test frontend
         run: yarn build
+
+      - name: Upload the
```
Comment (BenediktSeidl) [inline .github/workflows/ci.yml:26]: dist folder?

is this artifact actually used? i think we will need `actions/download-artifact@v3` to actually use it?

**Review by BenediktSeidl — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 7 above)
Comment (PhilippSiegmantel) [reply .github/workflows/ci.yml:26]: Not really, the only thing you can do for now is download it from the actions page. I thought it's at least someting for now.

I'd push actually using it in the CI to next week.

File: package.json [hunk 8]
```diff
@@ -14,19 +14,19 @@
   "license": "GPL-2.0-only",
```
Comment (BenediktSeidl) [inline package.json:14]: nothing changed here and the changes in `/yarn.lock` could also be removed?

**Review by BenediktSeidl — CHANGES_REQUESTED** [review]
one last thing. sorry, i somehow missed that you added another `package.json` file, so we could reduce the size of this commit.

**Review by BenediktSeidl — APPROVED** [review]

**Pull request merged**